src/main/java/core/levels/SpawnController.java

```diff
@@ -0,0 +1,25 @@
+package core.levels;
+
+import com.badlogic.gdx.math.Vector2;
+import core.world.WorldEntity;
+
+/**
+ * Controller class for spawning entities within the game
+ */
+public class SpawnController {
+
+    /**
+     * The x and y location where a WorldEntity should be spawned to
+     */
+    private Vector2 spawnLocation;
+
+    // TODO: Figure out SpawnController so it can just take in WorldEntity
+    public SpawnController(Vector2 spawnLocation){
```

Comment:
We can use the use case classes here. For ex. we could look up a character in the character manager based on its id and then spawn it following some condition.

Reply:
this will be changed to a command based system, not final